plugins/blocklog_plugin/blocklog_plugin.cpp

```diff
@@ -0,0 +1,140 @@
+
+#include <eosio/utilities/key_conversion.hpp>
+#include <eosio/utilities/common.hpp>
+
+#include <boost/algorithm/string.hpp>
+#include <boost/lexical_cast.hpp>
+
+#include <fc/io/json.hpp>
+#include <fc/variant.hpp>
+#include <signal.h>
+
+namespace eosio {
+using namespace chain;
+
+static appbase::abstract_plugin& _bnet_plugin = app().register_plugin<blocklog_plugin>();
+
+
+using namespace eosio;
+using namespace eosio::chain;
+using namespace eosio::chain::config;
+using fc::flat_map;
+
+struct blocklog_plugin_impl {
+   blocklog_plugin_impl()
+   {}
+
+   void read_log();
+
+   bfs::path                        blocks_dir;
+   bfs::path                        output_file;
+   fc::optional<block_log>          block_logger;
+   uint32_t                         first_block;
+   uint32_t                         last_block;
+};
+
+void blocklog_plugin_impl::read_log() {
+   block_logger.emplace(block_log(blocks_dir));
+   auto end = block_logger->read_head();
+   EOS_ASSERT( end, block_log_exception, "No blocks found in block log" );
+   EOS_ASSERT( end->block_num() > 1, block_log_exception, "Only one block found in block log" );
+
+   auto end_time = end->timestamp.to_time_point();
+   ilog( "existing block log with ${n} blocks", ("n",end->block_num()) );
+
+   std::ofstream output_blocks;
+   std::ostream* out;
+   if (!output_file.empty()) {
+      output_blocks.open(output_file.generic_string().c_str());
+      if (output_blocks.fail()) {
+         std::ostringstream ss;
+         ss << "Unable to open file '" << output_file.string() << "'";
+         throw std::runtime_error(ss.str());
+      }
+      out = &output_blocks;
+   }
+   else
+      out = &std::cout;
+
+   uint32_t block_num = (first_block < 1) ? 1 : first_block;
+   const uint32_t end_block = (last_block < end->block_num() ? last_block : end->block_num()) + 1;
+   signed_block_ptr next;
+   while((block_num < end_block) && (next = block_logger->read_block_by_num( block_num ))) {
+      fc::variant pretty_output;
+      abi_serializer::to_variant(*next,
+                                 pretty_output,
+                                 []( account_name n ) { return optional<abi_serializer>(); },
+                                 fc::microseconds(1000 * 1000));
+      const uint32_t ref_block_prefix = next->id()._hash[1];
+      auto enhanced_object = fc::mutable_variant_object
+                 ("block_num",next->block_num())
+                 ("id", next->id())
+                 ("ref_block_prefix", ref_block_prefix)
+                 (pretty_output.get_object());
+      *out << fc::json::to_pretty_string(enhanced_object) << "\n";
+      ++block_num;
+   }
+}
+
+blocklog_plugin::blocklog_plugin()
+:my(new blocklog_plugin_impl()) {
+}
+
+blocklog_plugin::~blocklog_plugin(){}
+
+void blocklog_plugin::set_program_options(options_description& cli, options_description& cfg)
+{
+   cfg.add_options()
+         ("blocks-dir,d", bpo::value<bfs::path>()->default_value("blocks"),
```

Comment:
-d is not unique, collides with --data-dir

Reply:
will remove